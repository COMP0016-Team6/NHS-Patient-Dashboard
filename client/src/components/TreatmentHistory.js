import React from "react";
import { useSelector } from "react-redux";

import { Table, TableHead, TableBody, TableCell, TableContainer, TableRow, Paper}  from '@material-ui/core';
import { useStylesTable } from '../styles/styles';

const TreatmentHistory = () => {
  const treatmentPlan = useSelector(state => state.patientPlan);
  const classes = useStylesTable();

  return (
    <TableContainer component={Paper} >
      <Table className={classes.table} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell align="center">Treatment Plan Description</TableCell>
            <TableCell align="center">Target Fluid (ml^3)</TableCell>
            <TableCell align="center">Target Energy (kcal)</TableCell>
            <TableCell align="center">Modified On</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {treatmentPlan.map(plan => (
          <TableRow key={plan.modified_time + plan.description} style={{overflow: "scroll"}}>
            <TableCell align="center">{plan.description}</TableCell>
            <TableCell align="center">{plan.target_feed_fluid}</TableCell>
            <TableCell align="center">{plan.target_feed_energy}</TableCell>
            <TableCell align="center">{plan.modified_time.toLocaleString()}</TableCell>
          </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  )
}

export default TreatmentHistory;