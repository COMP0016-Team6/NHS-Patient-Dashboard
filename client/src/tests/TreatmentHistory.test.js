import React from "react";
import TreatmentHistory from "../components/TreatmentHistory";
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { patientWrapper } from "./util";

it ("TreatmentHistory snapshot matches", () => {
  const tree = render(patientWrapper(<BrowserRouter><TreatmentHistory /></BrowserRouter>));
  expect(tree).toMatchSnapshot();
});
