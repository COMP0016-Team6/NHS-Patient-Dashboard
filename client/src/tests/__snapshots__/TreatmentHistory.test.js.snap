// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreatmentHistory snapshot matches 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <table
          aria-label="simple table"
          class="MuiTable-root makeStyles-table-1"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
              >
                Treatment Plan Description
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
              >
                Target Fluid (ml^3)
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
              >
                Target Energy (kcal)
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
              >
                Modified On
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
              style="overflow: scroll;"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
              >
                test
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
              >
                1.5
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
              >
                2.5
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
              >
                2020-05-05 20:20:20
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        aria-label="simple table"
        class="MuiTable-root makeStyles-table-1"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              Treatment Plan Description
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              Target Fluid (ml^3)
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              Target Energy (kcal)
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              Modified On
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
            style="overflow: scroll;"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              test
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              1.5
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              2.5
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              2020-05-05 20:20:20
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
